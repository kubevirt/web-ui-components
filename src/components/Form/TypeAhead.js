import React from 'react';
import PropTypes from 'prop-types';
import { TypeAheadSelect, noop } from 'patternfly-react';
import { isArray } from 'lodash';

export const TypeAhead = ({
  id,
  labelKey,
  defaultValue,
  placeholder,
  choices,
  multiple,
  selected,
  onChange,
  onBlur,
  disabled,
}) => {
  if (selected) {
    selected = isArray(selected) ? selected : [selected];
  }
  return (
    <TypeAheadSelect
      id={id}
      key={id}
      labelKey={labelKey}
      defaultInputValue={defaultValue}
      placeholder={placeholder}
      options={choices}
      multiple={multiple}
      selected={selected}
      onChange={onChange}
      onBlur={onBlur}
      disabled={disabled}
    />
  );
};

TypeAhead.defaultProps = {
  labelKey: undefined,
  defaultValue: undefined,
  placeholder: undefined,
  multiple: false,
  selected: undefined,
  onChange: noop,
  onBlur: noop,
  disabled: false,
};

TypeAhead.propTypes = {
  id: PropTypes.string.isRequired,
  labelKey: PropTypes.string,
  defaultValue: PropTypes.string,
  placeholder: PropTypes.string,
  choices: PropTypes.array.isRequired,
  multiple: PropTypes.bool,
  selected: PropTypes.oneOfType([PropTypes.array, PropTypes.object, PropTypes.string]),
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
  disabled: PropTypes.bool,
};
