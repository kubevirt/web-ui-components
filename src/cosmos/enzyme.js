import createContext from 'react-cosmos-test/enzyme';

import paths from '../../config/paths';

export default function createTestContext(args) {
  args.proxies = args.proxies || []; // workaround of react-cosmos-test bug
  return createContext({
    ...args,
    cosmosConfigPath: `${paths.config}/cosmos.config.js`,
  });
}
