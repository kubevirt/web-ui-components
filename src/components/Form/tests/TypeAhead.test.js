import React from 'react';
import { shallow } from 'enzyme';

import { TypeAhead } from '../TypeAhead';

const testTypeAheadContol = defaultValue => (
  <TypeAhead
    id="typeahead-1"
    defaultValue={defaultValue}
    placeholder="Select a record..."
    labelKey="name"
    choices={[{ id: 1, name: 'Record 1' }, { id: 2, name: 'Record 2' }]}
    onChange={() => {}}
    onBlur={() => {}}
  />
);

describe('<TextAreaControl />', () => {
  it('renders correctly', () => {
    const component = shallow(testTypeAheadContol());
    expect(component).toMatchSnapshot();
  });
  it('renders a default value', () => {
    const component = shallow(testTypeAheadContol('Record 1'));
    expect(component).toMatchSnapshot();
  });
});
