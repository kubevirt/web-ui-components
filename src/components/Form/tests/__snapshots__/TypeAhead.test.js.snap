// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextAreaControl /> renders a default value 1`] = `
<OnClickOutside(TypeaheadContainer(WrappedTypeahead))
  defaultInputValue="Record 1"
  disabled={false}
  eventTypes={
    Array [
      "mousedown",
      "touchstart",
    ]
  }
  excludeScrollbar={false}
  id="typeahead-1"
  key="typeahead-1"
  labelKey="name"
  multiple={false}
  onBlur={[Function]}
  onChange={[Function]}
  options={
    Array [
      Object {
        "id": 1,
        "name": "Record 1",
      },
      Object {
        "id": 2,
        "name": "Record 2",
      },
    ]
  }
  outsideClickIgnoreClass="ignore-react-onclickoutside"
  placeholder="Select a record..."
  preventDefault={false}
  stopPropagation={false}
/>
`;

exports[`<TextAreaControl /> renders correctly 1`] = `
<OnClickOutside(TypeaheadContainer(WrappedTypeahead))
  disabled={false}
  eventTypes={
    Array [
      "mousedown",
      "touchstart",
    ]
  }
  excludeScrollbar={false}
  id="typeahead-1"
  key="typeahead-1"
  labelKey="name"
  multiple={false}
  onBlur={[Function]}
  onChange={[Function]}
  options={
    Array [
      Object {
        "id": 1,
        "name": "Record 1",
      },
      Object {
        "id": 2,
        "name": "Record 2",
      },
    ]
  }
  outsideClickIgnoreClass="ignore-react-onclickoutside"
  placeholder="Select a record..."
  preventDefault={false}
  stopPropagation={false}
/>
`;
